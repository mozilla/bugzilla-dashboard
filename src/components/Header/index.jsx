import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';
import CredentialsMenu from '../../views/CredentialsMenu';

const styles = theme => ({
  styledToolbar: {
    display: 'flex',
    justifyContent: 'space-between',
    'min-height': theme.spacing.unit * 1,
  },
});

const Header = ({ classes, selectedTabIndex, handleTabChange }) => (
  <AppBar position="static">
    <Toolbar className={classes.styledToolbar}>
      <Tabs value={selectedTabIndex} onChange={handleTabChange}>
        <Tab label="Reportees" />
        <Tab label="Teams" />
        <Tab label="Components" />
      </Tabs>
      <CredentialsMenu />
    </Toolbar>
  </AppBar>
);

Header.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  selectedTabIndex: PropTypes.number.isRequired,
  handleTabChange: PropTypes.func.isRequired,
};

export default withStyles(styles)(Header);
