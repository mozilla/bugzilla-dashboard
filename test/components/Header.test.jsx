import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../../src/components/Header';

it('renders the reportees tab', () => {
  const tree = renderer
    .create((
      <Router>
        <Header
          selectedTabIndex={0}
          handleTabChange={() => null}
        />
      </Router>
    ))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
