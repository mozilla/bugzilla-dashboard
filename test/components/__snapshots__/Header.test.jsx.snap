// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the reportees tab 1`] = `
<header
  className="MuiPaper-root-11 MuiPaper-elevation4-17 MuiAppBar-root-2 MuiAppBar-positionStatic-6 MuiAppBar-colorPrimary-9"
>
  <div
    className="MuiToolbar-root-38 MuiToolbar-regular-40 MuiToolbar-gutters-39 Header-styledToolbar-1"
  >
    <div
      className="MuiTabs-root-42"
    >
      <div
        className="MuiTabs-flexContainer-43"
      >
        <div
          className="MuiTabs-scroller-45 MuiTabs-fixed-46"
          onScroll={[Function]}
          role="tablist"
          style={
            Object {
              "marginBottom": 0,
            }
          }
        >
          <div
            className="MuiTabs-flexContainer-43"
          >
            <a
              aria-current={null}
              aria-selected={true}
              className="MuiButtonBase-root-63 MuiTab-root-51 MuiTab-textColorInherit-53 MuiTab-selected-56"
              href="/reportees"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
            >
              <span
                className="MuiTab-wrapper-59"
              >
                <span
                  className="MuiTab-labelContainer-60"
                >
                  <span
                    className="MuiTab-label-61"
                  >
                    Reportees
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-96"
              />
            </a>
            <a
              aria-current={null}
              aria-selected={false}
              className="MuiButtonBase-root-63 MuiTab-root-51 MuiTab-textColorInherit-53"
              href="/teams"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
            >
              <span
                className="MuiTab-wrapper-59"
              >
                <span
                  className="MuiTab-labelContainer-60"
                >
                  <span
                    className="MuiTab-label-61"
                  >
                    Teams
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-96"
              />
            </a>
            <a
              aria-current={null}
              aria-selected={false}
              className="MuiButtonBase-root-63 MuiTab-root-51 MuiTab-textColorInherit-53"
              href="/components"
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="tab"
              tabIndex="0"
            >
              <span
                className="MuiTab-wrapper-59"
              >
                <span
                  className="MuiTab-labelContainer-60"
                >
                  <span
                    className="MuiTab-label-61"
                  >
                    Components
                  </span>
                </span>
              </span>
              <span
                className="MuiTouchRipple-root-96"
              />
            </a>
          </div>
          <span
            className="MuiPrivateTabIndicator-root-66 MuiPrivateTabIndicator-colorSecondary-68 MuiTabs-indicator-50"
            style={
              Object {
                "left": 0,
                "width": 0,
              }
            }
          />
        </div>
      </div>
    </div>
    <button
      className="MuiButtonBase-root-63 MuiButton-root-70 MuiButton-contained-81 MuiButton-containedSecondary-83 MuiButton-raised-84 MuiButton-raisedSecondary-86 MuiButton-sizeSmall-93 CredentialsMenu-button-69"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-71"
      >
        Sign in
      </span>
      <span
        className="MuiTouchRipple-root-96"
      />
    </button>
  </div>
</header>
`;
